pkg/build/controller/buildimagechangetrigger/buildimagechangetrigger_controller.go

```diff
@@ -0,0 +1,293 @@
+				continue
+			}
+
+			fromNamespace := from.Namespace
+			if len(fromNamespace) == 0 {
+				fromNamespace = config.Namespace
+			}
+
+			// only trigger a build if this image stream matches the name and namespace of the stream ref in the build trigger
+			// also do not trigger if the imagerepo does not have a valid DockerImageRepository value for us to pull
+			// the image from
+			if len(stream.Status.DockerImageRepository) == 0 || fromStreamName != stream.Name || fromNamespace != stream.Namespace {
+				continue
+			}
+
+			// This split is safe because ImageStreamTag names always have the form
+			// name:tag.
+			latestReference, ok = imageapi.ResolveLatestTaggedImage(stream, tag)
+			if !ok {
+				glog.V(4).Infof("unable to find tagged image: no image recorded for %s/%s:%s", stream.Namespace, stream.Name, tag)
+				continue
+			}
+			glog.V(4).Infof("Found ImageStream %s/%s with tag %s", stream.Namespace, stream.Name, tag)
+
+			// (must be different) to trigger a build
+			last := trigger.ImageChange.LastTriggeredImageID
+			next := latestReference
+
+			if len(last) == 0 || (len(next) > 0 && next != last) {
+				triggeredImage = next
+				shouldBuild = true
+				// it doesn't really make sense to have multiple image change triggers any more,
+				// so just exit the loop now
+				break
+			}
+		}
+
+		if shouldBuild {
+			glog.V(4).Infof("Running build for BuildConfig %s/%s", config.Namespace, config.Name)
+			buildCauses := []buildapi.BuildTriggerCause{}
+			// instantiate new build
+			request := &buildapi.BuildRequest{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      config.Name,
+					Namespace: config.Namespace,
+				},
+				TriggeredBy: append(buildCauses,
+					buildapi.BuildTriggerCause{
+						Message: buildapi.BuildTriggerCauseImageMsg,
+						ImageChangeBuild: &buildapi.ImageChangeCause{
+							ImageID: latestReference,
+							FromRef: from,
+						},
+					},
+				),
+				TriggeredByImage: &kapi.ObjectReference{
+					Kind: "DockerImage",
+					Name: triggeredImage,
+				},
+				From: from,
+			}
+			// TODO - does bullet 11 of https://github.com/kubernetes/community/blob/master/contributors/devel/controllers.md apply here?
+			// if so, what is the "owner reference" in this context?
+			if _, err := bc.bcInstantiator.Instantiate(config.Namespace, request); err != nil {
+				if kerrors.IsConflict(err) {
+					utilruntime.HandleError(fmt.Errorf("unable to instantiate Build for BuildConfig %s/%s due to a conflicting update: %v", config.Namespace, config.Name, err))
+				} else {
+					utilruntime.HandleError(fmt.Errorf("error instantiating Build from BuildConfig %s/%s: %v", config.Namespace, config.Name, err))
+				}
+				hasError = true
+				continue
+			}
+		}
+	}
+	if hasError {
+		return fmt.Errorf("an error occurred processing 1 or more build configurations; the image change trigger for image stream %s will be retried", stream.Status.DockerImageRepository)
```

Comment:
> Cataloging when hasError gets set, at first blush, the build instantiate error on conflict path still seems like a candidate for retrying immediately, no?

yeah that makes sense.

> But for the generic error case, it is less assured if our retry would help, and thus let's wait for the relist.

that means we should reverse the logic.  Instead of having fatalerrors that don't get retried immediately, we should have retryable errors that do get retried immediately (and a default behavior of not retrying).